### PR TITLE
tag.Do: use tag.Map from provided context

### PR DESCRIPTION
**Pull request by ramonza**
Fixes: #296

### CODE REVIEW

File: internal/readme/tags.go [hunk 1]
```diff
@@ -77,7 +77,8 @@ func tagsExamples() {
 	if err != nil {
 		log.Fatal(err)
 	}
-	tag.Do(ctx, tagMap, func(ctx context.Context) {
+	ctx = tag.NewContext(ctx, tagMap)
```
Comment (rakyll) [inline internal/readme/tags.go]: Could you run internal/readme/mkdocs.sh to regenerate the README please?

File: tag/example_test.go [hunk 2]
```diff
@@ -82,3 +82,19 @@ func ExampleFromContext() {
...
+	tagMap, err := tag.NewMap(ctx,
+		tag.Insert(key, "macOS-10.12.5"),
+		tag.Upsert(key, "macOS-10.12.7"),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ctx = tag.NewContext(ctx, tagMap)
+
```
Comment (rakyll) [inline tag/example_test.go]: Want to delete the arbitrary line here?

File: tag/map_test.go [hunk 3]
```diff
@@ -40,6 +40,38 @@ func TestContext(t *testing.T) {
...
+	Do(ctx, func(ctx context.Context) {
+			called = true
+	})
+
+	if !called {
+		t.Errorf("function should be called")
+	}
+}
+
+func TestDo(t *testing.T) {
```
Comment (rakyll) [inline tag/map_test.go]: We can merge TestDoFunctionCalled into this test case.

File: tag/map.go [hunk 4]
```diff
@@ -169,14 +169,16 @@ func NewMap(ctx context.Context, mutator ...Mutator) (*Map, error) {
 	return m, nil
 }
 
-// Do is similar to pprof.Do.
+// Do is similar to pprof.Do: a convenience for installing the tags
+// from the context as Go profiler labels. This allows you to
+// correlated runtime profiling with stats.
```
Comment (odeke-em) [inline tag/map.go]: s/correlated/correlate/

**Pull request merged**